data/data/azure/main.tf

```diff
@@ -123,18 +130,26 @@ resource "azurerm_role_assignment" "main" {
 # https://github.com/MicrosoftDocs/azure-docs/issues/13728
 resource "azurerm_dns_zone" "private" {
   name                           = var.cluster_domain
-  resource_group_name            = azurerm_resource_group.main.name
```

Comment:
this change looks incorrect. the private zone should be in the resource group of the cluster and not network.

also removing `resolution_virtual_network_ids` means DNS is not being used....

Reply:
This was unintentional after changing to new DNS API and back. I have fixed it.